cocotb/handle.py

```diff
@@ -216,10 +216,15 @@ def _sub_handle_key(self, name):
         return name.split(".")[-1]
 
     def _getAttributeNames(self):
-        """Permits IPython tab completion to work."""
+        """Permits older IPython tab completion to work."""
         self._discover_all()
         return dir(self)
 
+    def __dir__(self):
+        """Permits newer IPython tab completion to work."""
+        self._discover_all()
+        return super().__dir__() + [str(k) for k in self._sub_handles]
```

Comment:
This won't work in python 2 - needs to be `super(RegionObject, self).__dir__()`

Reply:
Thanks, fixed.